src/main.rs

```diff
@@ -27,9 +30,44 @@ fn main() {
     let block_size = 1024;
 
     // Read audio data and write it to the output text file (one column per channel)
-    let mut out = File::create(&args[2]).expect("Unable to create file");
+    //let mut out = File::create(&args[2]).expect("Unable to create file");
+    let mut out = hound::WavWriter::create(&args[2], spec).unwrap();
+    let mut sample_index = 0;
+    let mut block_index: u32 = 0;
+    let mut input_buffer = vec![vec![f32::default(); channels as usize]; block_size as usize];
+    let mut output_buffer = vec![vec![f32::default(); channels as usize]; block_size as usize];
+    let mut comb_filter = CombFilter::new(FilterType::FIR, 1.0, 44100.0, channels as usize);
```

Comment:
I think the assignment asked us to take the comb filter parameters from command line, so maybe it would be good to assign the variables using &args[i] instead of hard coding it here